overlord/configstate/configcore/timezone.go

```diff
@@ -0,0 +1,86 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package configcore
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"regexp"
+
+	"github.com/snapcore/snapd/osutil"
+	"github.com/snapcore/snapd/overlord/configstate/config"
+)
+
+func init() {
+	// add supported configuration of this module
+	supportedConfigurations["core.system.timezone"] = true
+}
+
+var validTimezone = regexp.MustCompile(`^[a-zA-Z0-9+-]+(/[a-zA-Z0-9+-]+)?$`).MatchString
```

Comment:
How did you come up with this pattern?

Looking at /usr/share/zoneinfo/ I see a timezone called `Africa/Sao_Tome` that would not match this pattern. In addition the number of slashes may be greater than one, one example is `America/Argentina/Cordoba`.

Reply:
Thank you, indeed, I missed those ones.